File: basicAlgorithms/_slidingWindow.js
function slidingWindow(arr){
    let l = 0;
    let subset = [];
    let longest = 0;
    let w = 0;
    for(let r = 0;r<arr.length;r++){
        while(subset.indexOf(arr[r]) != -1){
            subset.shift();
            l++;
        }

        subset.push(arr[r]);

        w = (r-l) + 1;

        if(w>longest){
            longest = w;
        }

    }

    return longest;
}
let longest = slidingWindow(["p","w","w","k","e","w","x","x","z","a"]);

console.log(longest);

File: basicAlgorithms/_slidingWindow.py
def slidingWindow(str):
    l = 0
    w = 0
    subset = set()
    longest = 0
    
    for r in range(len(str)):
        while str[r] in subset:
            subset.pop()
            l = l + 1

        subset.add(str[r])
        w = (r-l) + 1

        if w>longest:
            longest = w

    return longest

print(slidingWindow(["p","w","w","k","e","w","x","x","z","a"]))